app/uk/gov/hmrc/helptosave/repo/EnrolmentStore.scala

```diff
@@ -144,11 +153,29 @@ class MongoEnrolmentStore @Inject() (mongo:   ReactiveMongoComponent,
     }
     )
   }
+
+  override def insert(nino:              NINO,
+                      itmpFlag:          Boolean,
+                      eligibilityReason: Option[Int],
+                      channel:           String)(implicit hc: HeaderCarrier): EitherT[Future, String, Unit] =
+    EitherT(
+      doInsert(nino, eligibilityReason, channel, itmpFlag)
+        .map[Either[String, Unit]] { writeResult ⇒
+          if (writeResult.writeErrors.nonEmpty) {
+            Left(writeResult.writeErrors.map(_.errmsg).mkString(","))
+          } else {
+            Right(())
+          }
+        }.recover {
+          case e ⇒
+            Left(e.getMessage)
+        }
+    )
 }
 
 object MongoEnrolmentStore {
 
-  private[repo] case class EnrolmentData(nino: String, itmpHtSFlag: Boolean)
+  private[repo] case class EnrolmentData(nino: String, itmpHtSFlag: Boolean, eligibilityReason: Option[Int] = None, channel: Option[String] = None)
```

Comment:
this is used for reading from mongo and both fields are optional here just to make it compatible with existing records as those have only `nino` and `itmpFlag` fields